module/variable.go

```diff
@@ -0,0 +1,18 @@
+package module
+
+import (
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type Variable struct {
+	Description     string
+	Type            cty.Type
+	TypeConstraints schema.ExprConstraints
```

Comment:
What do we need `TypeConstraints` here for? 🤔

Reply:
ExprConstraints will be needed when you specify the schema you can derive them from the cty.Type but making this conversion every time we need that could be expensive that is why I thought about being able to store this in the variables.

Reply:
I think that creating one slice with a single element is negligible when we have to modify the schema anyway? I don't mind adding it if we conclude that it makes a _significant_ impact on the UX/CPU/memory, but I'd prefer to start with a simpler and cleaner API and work from there, rather than "optimize early", if possible.

Reply:
Perfect will remove and we can optimize it later.